okhttp/src/main/kotlin/okhttp3/internal/cache/CacheLock.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2024 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.cache
+
+import android.annotation.SuppressLint
+import java.io.Closeable
+import java.nio.channels.FileChannel
+import java.nio.file.StandardOpenOption
+import java.util.Collections
+import okhttp3.internal.platform.Platform
+import okio.FileSystem
+import okio.Path
+import org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
+
+internal object CacheLock {
+  private val openCaches = Collections.synchronizedMap(mutableMapOf<Path, Exception>())
+
+  fun openLock(
+    fileSystem: FileSystem,
+    directory: Path,
+  ): Closeable {
+    return if (fileSystem == FileSystem.SYSTEM && !Platform.isAndroid) {
```

Comment:
Ooooh this is a pretty loud request for Okio to support locks!

Reply:
https://github.com/square/okio/issues/1464